secure_message/repository/retriever.py

```diff
@@ -195,6 +197,49 @@ def retrieve_thread_list(user, request_args):
 
         return result
 
+    @staticmethod
+    def _retrieve_internal_thread_list(request_args, user):
+        """Retrieve a list of threads for an internal user"""
+        conditions = []
+
+        logger.info("Retrieving list of threads for internal user", user_uuid=user.user_uuid)
+
+        if request_args.ru_id:
+            conditions.append(SecureMessage.ru_id == request_args.ru_id)
+
+        if request_args.survey:
+            conditions.append(SecureMessage.survey == request_args.survey)
+
+        if request_args.cc:
+            conditions.append(SecureMessage.collection_case == request_args.cc)
+
+        if request_args.ce:
+            conditions.append(SecureMessage.collection_exercise == request_args.ce)
+
+        try:
+            t = db.session.query(SecureMessage.thread_id, func.max(Status.id)  # pylint:disable=no-member
+                                 .label('status_id')) \
+                .join(Events).join(Status) \
+                .filter(or_(and_(SecureMessage.from_internal == False, Status.label == Labels.INBOX.value),  # NOQA
+                            and_(SecureMessage.from_internal == True,
+                                 Status.label.in_([Labels.SENT.value, Labels.DRAFT.value]))
+                            )
+                        ) \
+                .group_by(SecureMessage.thread_id).subquery('t')
+
+            conditions.append(SecureMessage.thread_id == t.c.thread_id)
+            conditions.append(Status.id == t.c.status_id)
+
+            result = SecureMessage.query.join(Events).join(Status) \
+                .filter(and_(*conditions)) \
+                .order_by(t.c.status_id.desc()).paginate(request_args.page, request_args.limit, False)
+
+        except Exception as e:
+            logger.exception('Error retrieving messages from database')
```

Comment:
Any reason for catching Exception rather than the top level SQLAlchamey exception?

Reply:
Nope , it should be more specific . Same issue on other get list type endpoints . Changing those also.